pkg/client/metrics/prometheus/prometheus.go

```diff
@@ -59,6 +59,9 @@ type latencyAdapter struct {
 }
 
 func (l *latencyAdapter) Observe(verb string, u url.URL, latency time.Duration) {
+	if l.m == nil {
```

Comment:
I don't think this makes sense. If the latencyAdapter isn't already registered, then what how is the object which is throwing the error being initialized and this Observe function called in the first place?

Reply:
I did a thorough search to most recent [prometheus/client-golang](https://github.com/prometheus/client_golang) codebase, and was able to find the root cause.

The error happens in this [function](https://github.com/kubernetes/kubernetes/blob/master/vendor/github.com/prometheus/client_golang/prometheus/histogram.go#L337). The most recent version has the [fix](https://github.com/prometheus/client_golang/blob/master/prometheus/histogram.go#L465-L469) already, but we didn't update the file.

Reply:
@cheftako @caesarxuchao

Reply:
Am I misunderstanding something? That fix doesn't fix anything, it will just panic instead of the nil error.